### PR TITLE
Return error code on failure with occ app:enable/disable

### LINKED ISSUE #17980: occ app:enable and app:disable don't return error code in case of failure
### Steps to reproduce
1. Try to enable or disable a non-existing app -> `occ app:enable random_app` and / or `occ app:disable random_app`

### Expected behaviour
The command returns an error code so it's easier to script with it

### Actual behaviour
The command exits fine

### Server configuration
**Operating system**: ubuntu 14.04

**Web server:** apache 2.4.7

**Database:** mysql

**PHP version:** 5.5.9

**ownCloud version:** OC 8.1.0

**Updated from an older ownCloud or fresh install:** fresh install

### Notes
It's likely to happen with more versions and other commands

### PR DESCRIPTION
**Pull request by PVince81**
Fixes https://github.com/owncloud/core/issues/17980

@jvillafanez here you go, happy ?

Please review/test @DeepDiver1975 @Xenopathic @icewind1991 @nickvergessen

### CODE REVIEW

**Comment by nickvergessen** [comment]
:+1: 

Add to the documentation?
https://github.com/owncloud/documentation/blob/167c42d0a29d9d9e3c2d80b7724ad64cb09e7859/admin_manual/configuration_server/occ_command.rst#apps-commands

File: core/command/app/disable.php [hunk 1]
```diff
@@ -47,9 +47,11 @@ protected function execute(InputInterface $input, OutputInterface $output) {
 				$output->writeln($appId . ' disabled');
 			} catch(\Exception $e) {
```
Comment (Xenopathic) [inline core/command/app/disable.php]: Why do we catch the exception at all? Symfony Console will print the exception message, and will return with a non-zero exit code.

Comment (PVince81) [inline core/command/app/disable.php]: no idea...

**Comment by jvillafanez** [comment]
Code is working, so unless there is a change due to @Xenopathic 's comment :+1: 

I guess that we're catching the exceptions since the exit code will depend on the code of the exception. I think the default code is 0 unless stated otherwise, so probably the return code of the occ command will be 0 for the most of the cases in case of exception.

File: core/command/app/disable.php (hunk 1 above)
Comment (nickvergessen) [inline core/command/app/disable.php]: @Xenopathic because of reasons :smile:

**Comment by PVince81** [comment]
@nickvergessen I don't think this needs to be documented, this is how every command should work. We should rather make sure that all commands return 0 on success and non-zero on error. Unless you want every type of error to be documented separately ?

**Comment by DeepDiver1975** [comment]
> Unless you want every type of error to be documented separately ?

not really

**Comment by DeepDiver1975** [comment]
> this is how every command should work. We should rather make sure that all commands return 0 on success and non-zero on error.

indeed

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
Bääääh - this is now breaking travis builds all over the place:
https://magnum.travis-ci.com/owncloud/enterprise_key/jobs/22213411

**Comment by DeepDiver1975** [comment]
so we either revert or declare enabling an enabled app as non error (same for disabling an disabled apps)

@PVince81

**Comment by PVince81** [comment]
Let's go with the non-error route. I'll submit a PR.

**Comment by DeepDiver1975** [comment]
> Let's go with the non-error route. I'll submit a PR.

THX

**Comment by PVince81** [comment]
Here: https://github.com/owncloud/core/pull/18415

**Pull request merged**